AI/VCAI/Goals.cpp

```diff
@@ -1125,7 +1131,7 @@ TGoalVec GatherArmy::getAllPossibleSubgoals()
 		{
 			for (auto h : cb->getAvailableHeroes(t)) //we assume that all towns have same set of heroes
 			{
-				if (h && h->getTotalStrength() > 500) //do not buy heroes with single creatures for GatherArmy
+				if (h && (h->getTotalStrength() > 500 && ai->freeResources()[Res::GOLD] > GameConstants::HERO_GOLD_COST) //do not buy heroes with single creatures for GatherArmy

```

Comment:
For now it's OK. In the future AI might want to collect gold for a hero even if it doesn't have it right now.